asv/environment.py

```diff
@@ -890,7 +890,9 @@ def _install_project(self, repo, commit_hash, build_dir):
             # Run pip via python -m pip, avoids shebang length limit on Linux.
             # Don't run it in build directory, because it may contain Python packages
             # that pip believes to be already installed.
-            cmd = ["in-dir={env_dir} python -mpip install {wheel_file}"]
+            # --force-reinstall is needed since versions may not change between
+            # asv runs (esp. for compare), e.g. gh-1421
+            cmd = ["in-dir={env_dir} python -mpip install {wheel_file} --force-reinstall"]
```

Comment:
`--force-reinstall` (basically the same as `pip uninstall && pip install` is used here, but `-I` might also be the "right thing to do" for testing commits as long as the files have not changed.

Reply:
I'd have also been open to simply documenting the fix described in https://github.com/airspeed-velocity/asv/issues/1417#issuecomment-2282754045 but it seems like an issue a lot of people might have.